src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/MobilePay/MobilePayPaymentOperations.cs

```diff
@@ -1,58 +1,56 @@
-
-namespace Swedbankpay.Sdk.Payments.MobilePayPayments
-{
-    public class MobilePayPaymentOperations : OperationsBase
-    {
-        public MobilePayPaymentOperations(OperationList operations, HttpClient client)
-        {
-            foreach (var httpOperation in operations)
-            {
-                switch (httpOperation.Rel.Value)
-                {
-                    case PaymentResourceOperations.UpdatePaymentAbort:
-                        Abort = async payload =>
-                            await client.SendAsJsonAsync<MobilePayPaymentResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.RedirectAuthorization:
-                        RedirectAuthorization = httpOperation;
-                        break;
-
-                    case PaymentResourceOperations.ViewAuthorization:
-                        ViewAuthorization = httpOperation;
-                        break;
-
-                    case PaymentResourceOperations.CreateCapture:
-                        Capture = async payload =>
-                            await client.SendAsJsonAsync<CaptureResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.CreateCancellation:
-                        Cancel = async payload =>
-                            await client.SendAsJsonAsync<CancellationResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-
-                    case PaymentResourceOperations.CreateReversal:
-                        Reverse = async payload =>
-                            await client.SendAsJsonAsync<ReversalResponse>(httpOperation.Method, httpOperation.Href, payload);
-                        break;
-                }
-                this.Add(httpOperation.Rel, httpOperation);
-
-            }
-        }
-        public Func<PaymentAbortRequest, Task<MobilePayPaymentResponse>> Abort { get; }
-        public Func<MobilePayPaymentCancelRequest, Task<CancellationResponse>> Cancel { get; }
-        public Func<MobilePayPaymentCaptureRequest, Task<CaptureResponse>> Capture { get; }
-        public HttpOperation RedirectAuthorization { get; }
-        public Func<MobilePayPaymentReversalRequest, Task<ReversalResponse>> Reverse { get; }
-        public HttpOperation ViewAuthorization { get; }
-    }
-}
+﻿using SwedbankPay.Sdk.Common;
+using SwedbankPay.Sdk.Extensions;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace SwedbankPay.Sdk.PaymentInstruments.MobilePay
+{
+    public class MobilePayPaymentOperations : OperationsBase, IMobilePayPaymentOperations
+    {
+        public MobilePayPaymentOperations(IOperationList operations, HttpClient client)
+        {
+            foreach (var httpOperation in operations)
+            {
+                switch (httpOperation.Rel.Value)

```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)